### ISSUE TITLE
Pipeline: Support set pod template for tasks

### ISSUE THREAD

**Issue by wuhuizuo** [tektoncd/pipeline#6742]
### Feature request

Support set pod template for tasks in `Pipeline` resource.

### Use case
I want a pipeline with two parallel tasks, one task run on AMD64 arch node and another run on ARM64 node.
Currently only PipelineRun support pod template to do it, but it's complex for my case, it should decide internal by params.

**Comment by lbernick** [comment]
@wuhuizuo would [matrix](https://github.com/tektoncd/pipeline/blob/main/docs/matrix.md) work for your use case?

**Comment by vdemeester** [comment]
> @wuhuizuo would [matrix](https://github.com/tektoncd/pipeline/blob/main/docs/matrix.md) work for your use case?

It doesn't as it doesn't "apply" to `PipelineRun` and in that particular case, `taskRunTemplate`/`taskRunspecs` (because this is where we can set topology constraint, …).

**Comment by wuhuizuo** [comment]
> @wuhuizuo would [matrix](https://github.com/tektoncd/pipeline/blob/main/docs/matrix.md) work for your use case?

No, it can not. that I want to do: set arch in params or matrix to make parallel tasks on different arch nodes.

**Comment by vdemeester** [comment]
@lbernick @jerop this is a "shortcoming" of the `matrix` work that is making it a bit less useful than I thought it would to be honest 🙃

**Comment by jerop** [comment]
@vdemeester would `matrix` at the `pipeline` level be more useful?

**Comment by vdemeester** [comment]
@jerop I don't think it would solve too much (or even what it would look like — except if it's a typo and you meant `pipelinerun` 😛). The main trick is, when you want to use `matrix` for some runtime information. The basic example that comes to mind is for multi-arch build on nodes with different architecture (on a heterogenous cluster). Today, we cannot do this with matrix, because those are runtime information, that cannot benefit from the `matrix` feature I believe.

**Comment by jerop** [comment]
yes, I meant `pipelinerun` 🙃 

pipeline declares string params, and pipelinerun provides array params to fan out to multiple child pipelineruns

seems to me that will help with the multi-arch use case, right?

**Comment by vdemeester** [comment]
> seems to me that will help with the multi-arch use case, right?

It helps if we use cross-compilation to do multi-arch build, but it doesn't if we need the "host" to be a certain architecture (as of today I mean).

**Comment by wuhuizuo** [comment]
> > seems to me that will help with the multi-arch use case, right?
> 
> It helps if we use cross-compilation to do multi-arch build, but it doesn't if we need the "host" to be a certain architecture (as of today I mean).

@vdemeester 
Sorry for late reply, not just for building, may be some tests(unit tests or integration tests ...) need run on different-arch nodes.

**Comment by wuhuizuo** [comment]
I would like to implement it. ❤️

**Comment by raballew** [comment]
@wuhuizuo Any update on your efforts to implement this feature?

**Comment by jwx0925** [comment]
Any update ?
In our case, we have changed the parallel matrix build tasks for amd64 and arm64 to two independent tasks.

**Comment by wuhuizuo** [comment]
have no time to finish the feature.

/unassign

**Comment by dorzel** [comment]
Currently working on this in the linked PR, appreciate any feedback/suggestions
